ql/src/java/org/apache/hadoop/hive/ql/DriverUtils.java

```diff
@@ -51,10 +51,11 @@ public static void runOnDriver(HiveConf conf, String user,
       Driver driver = new Driver(qs, user, null, null);
       driver.setCompactionWriteIds(writeIds, compactorTxnId);
       try {
-        CommandProcessorResponse cpr = driver.run(query);
-        if (cpr.getResponseCode() != 0) {
-          LOG.error("Failed to run " + query, cpr.getException());
-          throw new HiveException("Failed to run " + query, cpr.getException());
+        try {
+          driver.run(query);
+        } catch (CommandProcessorException e) {
+          LOG.error("Failed to run " + query, e.getException());
+          throw new HiveException("Failed to run " + query, e.getException());
```

Comment:
e.getException() => e
as e contains some descriptive message - what is the issue